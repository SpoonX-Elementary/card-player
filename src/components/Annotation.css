.cardContainer {
  width: 854px;
  height: 480px;
  background-color: lavenderblush;
  position: absolute;
  z-index: 10;
  display: flex;
  align-items: flex-start;
  justify-content: flex-end;
}

.exit {
  margin: 20px;
}

.exit:hover {
  text-decoration: underline;
  margin: 20px;
}
