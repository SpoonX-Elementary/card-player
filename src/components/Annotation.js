import React from 'react';
import PropTypes from 'prop-types';
import './Annotation.css';

export default class Annotation extends React.Component {
  render() {
    return this.props.hidden
      ? null
      : (
        <div className="cardContainer">
          <a className="exit" onClick={this.props.hideCard}>Close</a>
        </div>
      );
  }
}

Annotation.propTypes = {
  hidden: PropTypes.bool,
  hideCard: PropTypes.func.isRequired
};

Annotation.defaultProps = {
  hidden: false
};

