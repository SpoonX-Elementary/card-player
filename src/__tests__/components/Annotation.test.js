import React from 'react';
import renderer from 'react-test-renderer';
import Annotation from '../../components/Annotation';

test('Should render Annotation', () => {
  const tree = renderer
    .create(<Annotation hideCard={() => {}} hidden={false} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should not render Annotation', () => {
  const tree = renderer
    .create(<Annotation hideCard={() => {}} hidden />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
