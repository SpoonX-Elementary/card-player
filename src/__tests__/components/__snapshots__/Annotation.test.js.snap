// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not render Annotation 1`] = `null`;

exports[`Should render Annotation 1`] = `
<div
  className="cardContainer"
>
  <a
    className="exit"
    onClick={[Function]}
  >
    Close
  </a>
</div>
`;
